packages/bp/src/core/events/event-repository.ts

```diff
@@ -74,11 +73,19 @@ export class EventRepository {
     }
 
     const event = events[0]
+    const details = event.event.decision?.sourceDetails
```

Comment:
Try to avoid using the same variable name twice. Also, since the typing is not correct, you can cast it to the correct type. 

You don't need additional checks because in the findEvents query, you already ask only for "incoming" events.

```js
const storedEvent = events[0]
const incomingEvent = storedEvent as sdk.IO.IncomingEvent

const details = incomingEvent.decison?.sourceDetails
```